src/athena.hpp

```diff
@@ -160,28 +157,28 @@ enum class UserHistoryOperation {sum, max, min};
 // function pointer prototypes for user-defined modules set at runtime
 
 using BValFunc = void (*)(
-    MeshBlock *pmb, Coordinates *pco, AthenaArray<Real> &prim, FaceField &b,
+    MeshBlock *pmb, Coordinates *pco, ParArrayND<Real> &prim, FaceField &b,
```

Comment:
It doesn't hurt to pass `ParArrayND` by reference. But you don't need to, since they shallow copy.